kb/cluster-metrics-reference.md

```diff
@@ -0,0 +1,126 @@
+---
+slug: cluster-metrics-reference
+title: Cluster metrics reference
+tags:
+  - kb-article
+date: 2022-09-15T00:00:00Z
+---
+
+The Temporal Cluster reports a range of metrics to help operators get visibility into the Cluster’s performance and set up alerts.
+All metrics emitted by the Cluster are listed in [defs.go](https://github.com/temporalio/temporal/blob/master/common/metrics/defs.go).
+
+For details on setting up metrics in your Cluster configuraiton, see [Temporal Cluster](/references/configuration/##global).
+
+The [dashboards repository](https://github.com/temporalio/dashboards) contains community-driven Grafana dashboards that can be used for monitoring Temporal Server and SDK metrics.
+You can use these as a reference to build your own dashboards.
+For any metrics that are missing in the dashboards, use [defs.go](https://github.com/temporalio/temporal/blob/master/common/metrics/defs.go) as a reference.
+
+Note that apart from these metrics emitted by the Cluster, you should also monitor infrastructure-specific metrics like CPU, memory, and network for all hosts that are running Temporal Cluster services.
+
+## Metrics reference
+
+Temporal emits metrics for each gRPC service request.
+These metrics are emitted with `type`, `operation`, and `namespace` tags, which provide visibility into service usage and show the request rates across services, namespaces, and even operations.
+
+- Use the `operation` tag with each metric in your query to get request rates, error rates or latencies per operation.
+- Use `service_type` with the [service role tag values](https://github.com/temporalio/temporal/blob/bba148cf1e1642fd39fa0174423b183d5fc62d95/common/metrics/defs.go#L108) to get details for the specific service.
+
+For example, to get the Workflow Task start to schedule ratio, use:
+
+```
+sum(rate(service_requests{service_type="history", operation="RecordWorkflowTaskStarted"}[2m])) / sum(rate(service_requests{service_type="matching", operation="AddWorkflowTask"}[2m]))
```

Comment:
This is not a good example to look at, like we rarely looked at this start to schedule ratio.
A better example would be service requests by operation on frontend:
sum by (operation) (rate(service_requests{service_name="frontend"}[2m]))

Reply:
updated